project/src/main/java/com/google/sps/data/DataAccessor.java

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.repackaged.com.google.api.client.util.Strings;
+
+public class DataAccessor {
+
+  private final DatastoreService datastoreService;
+  @VisibleForTesting
+  final String userEntityName = "User";
+
+  DataAccessor() {
+    this.datastoreService = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @VisibleForTesting
+  DataAccessor(DatastoreService datastore) {
+    this.datastoreService = datastore;
+  }
+
+  /**
+  * @param userId the id of the user that we want to check the registration status about
+  * @return whether the user is registered to our system, by checking whether their id was
+  *     previously added to datastore.
+  */
+  public boolean isRegisteredId(String userId) {
+    Key userIdKey = KeyFactory.createKey(userEntityName, userId);
+    Filter userIdFilter =
+        new Query.FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL, userIdKey);
+    Query query = new Query(userEntityName).setFilter(userIdFilter).setKeysOnly();
+    PreparedQuery results = datastoreService.prepare(query);
+    return results.asList(FetchOptions.Builder.withDefaults()).size() > 0;
+  }
+
+  /**
+  * Registers the user in our system, by adding an entity that represents them to datastore.
+  *
+  * @param userId the id of the user that we want to check the registration status about
+  */
+  public void registerUserId(String userId) {
+    if (Strings.isNullOrEmpty(userId)) {
```

Comment:
That's a great question. 
The registration servlet won't return anything, like whether the registration was successful, the user was already registered, etc, because it's not required and this information won't be used.  And this happens because this information has no direct implications on the user or what we show him.
This can be empty if there was a problem with handling the user id in the functions that call this function (for example the token wasn't verified successfully). Your comment made me think that maybe we shouldn't return here, but throw an error. what do you think?

Reply:
Yes I think that an error should be thrown if the user tried to register and for some reason didn't succeed, but maybe in earlier stages and not here.

And a question since you brought it up - in the next milestone we will need to know out side of the registration Servlet if the user is registered in order to show him the old/new option, etc. I didn't find in the doc how it's done if there is no output from the servlet (we can also discuss this 1:1 when we get to it, just got a little confused).

Reply:
Yes that's an interesting point, yet it's unrelated to this PR, we'll discuss it 1:1 :)
About the error - I'm wondering if it should be an invalid input (InvalidArgumentException) in **this function**. I think it should. I'm changing this. Please LMK if you think otherwise.

Reply:
Sounds good